### PR TITLE
add individual build for windows and run from c: drive (more disk space)

### PR DESCRIPTION
**Pull request by OyvindLGjesdal**

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -1,13 +1,7 @@
...
 
 name: Apache Jena CI
-
-on:
-  push:
-    branches: [ main ]
-#  pull_request:
-#    branches: [ main ]
-
+on: workflow_dispatch
```
Comment (kinow) [inline .github/workflows/maven.yml:4]: Doesn't it make every other build, even linux, triggered manually only?

Comment (OyvindLGjesdal) [reply .github/workflows/maven.yml:4]: Yes. I think there has to be be individual workflow files maven.yml (linux, always run on main?), maven_windows.yml?, maven_macos.yml?, if only some  are to be triggered manually.

Comment (kinow) [reply .github/workflows/maven.yml:4]: I think so. At the project I am working on at $work, we have a [separate workflow](https://github.com/cylc/cylc-flow/blob/b5955ab1c2b16cc0a76baaf5826748bb8c6e289f/.github/workflows/1_create_release_pr.yml) for some release automation. Maybe we can have a `maven_windows.yml` or `windows.yml`. The rest of the file looks good after a cursory look :+1:

**Review by kinow — COMMENTED** [review]
New workflow looks good. @OyvindLGjesdal I think you triggered a build in your fork: https://github.com/OyvindLGjesdal/jena/actions/runs/615560102

I'm just waiting for that to pass to approve here :+1:

**Comment by kinow** [comment]
@OyvindLGjesdal I guess these errors were expected?

```
2021-03-02T22:08:51.9281214Z [ERROR] Tests run: 189, Failures: 1, Errors: 1, Skipped: 2, Time elapsed: 20.131 s <<< FAILURE! - in org.apache.jena.fuseki.TS_FusekiWebapp
2021-03-02T22:08:51.9283606Z [ERROR] task_6(org.apache.jena.fuseki.TestAdmin)  Time elapsed: 0.001 s  <<< FAILURE!
2021-03-02T22:08:51.9284723Z java.lang.AssertionError
2021-03-02T22:08:51.9285757Z 	at org.apache.jena.fuseki.TestAdmin.task_6(TestAdmin.java:485)
2021-03-02T22:08:51.9286540Z 
2021-03-02T22:08:51.9288445Z [ERROR] task_7(org.apache.jena.fuseki.TestAdmin)  Time elapsed: 1.421 s  <<< ERROR!
2021-03-02T22:08:51.9290241Z org.apache.jena.atlas.web.HttpException: 400 - Bad Request
2021-03-02T22:08:51.9292469Z 	at org.apache.jena.fuseki.TestAdmin.execSleepTask(TestAdmin.java:577)
2021-03-02T22:08:51.9296451Z 	at org.apache.jena.fuseki.TestAdmin.task_7(TestAdmin.java:491)
2021-03-02T22:08:51.9297289Z 
2021-03-02T22:08:52.3319544Z [INFO] 
2021-03-02T22:08:52.3322402Z [INFO] Results:
2021-03-02T22:08:52.3322971Z [INFO] 
2021-03-02T22:08:52.3323816Z [ERROR] Failures: 
2021-03-02T22:08:52.3324422Z [ERROR]   TestAdmin.task_6:485
2021-03-02T22:08:52.3325270Z [ERROR] Errors: 
2021-03-02T22:08:52.3326900Z [ERROR]   TestAdmin.task_7:491->execSleepTask:577 � Http 400 - Bad Request
2021-03-02T22:08:52.3327817Z [INFO] 
2021-03-02T22:08:52.3328408Z [ERROR] Tests run: 189, Failures: 1, Errors: 1, Skipped: 2
```

Probably for a follow-up issue, and not a blocker for this PR?

**Comment by OyvindLGjesdal** [comment]
I looked at getting the windows (and macOs) worflow to run/pass earlier, and I think it is related to https://github.com/apache/jena/pull/941
 
With [some changes to timeouts](https://github.com/OyvindLGjesdal/jena/compare/jena-2052-workflow-add-windows-and-dispatch...OyvindLGjesdal:windows-timeout) to the tests, the [windows build passes](https://github.com/OyvindLGjesdal/jena/runs/2017540130?check_suite_focus=true) sometimes, but I'm not sure if the tests are still meaningful/testing the correct thing.

**Comment by afs** [comment]
Clearly these timeout tests need rewriting. Some can switch to use `Awaitability`. Others don't seem to fit the patterns provided. They are expecting exceptions after a certain time has elapsed while some work is being done.

**Review by afs — APPROVED** [review]

**Comment by OyvindLGjesdal** [comment]
I've not updated the PR in writing as ready, since the macOs job seems to fail, even if the windows job passes, so I was unsure of if this was related to the changed test, but looking at the failing jobs, this seems to be unrelated to these changes.

https://github.com/OyvindLGjesdal/jena/runs/2113156358?check_suite_focus=true

 It was cool to try out the awaitility library/dsl.

**Comment by afs** [comment]
The MacOS subjob is unreliable - it works sometimes, usually when the system is quiet. It fails in the same place on a timeout. It takes a lot longer than the Linux job.

I suspect that the machines used are very slow to due high usage. 

To make progress, I suggest we get this PR in to get the Windows build working and begin to split the work into separate jobs.

The project isn't yet depending on github actions to succeed, or not. The jenkins suite of jobs covers several JDKs, and Windows. No MacOS coverage but one pass sometime on GH actions is enough to show that the code works.

Separately, we can tune the timeouts for MacOS somehow and get it working.

**Pull request merged**